azkaban-common/src/main/java/azkaban/trigger/TriggerManager.java

```diff
@@ -257,10 +255,17 @@ private void checkAllTriggers() throws TriggerManagerException {
           scannerStage = "Checking for trigger " + t.getTriggerId();
 
           if (t.getStatus().equals(TriggerStatus.READY)) {
-            if (t.triggerConditionMet()) {
+
+            /**
+             * Prior to this change, expiration condition should never be called though
+             * we have some related code here. ExpireCondition used the same BasicTimeChecker
+             * as triggerCondition do. As a consequence, we need to figure out a way to distinguish
+             * the previous ExpireCondition and this commit's ExpireCondition.
+             */
+            if (t.getExpireCondition().getExpression().contains("EndTimeChecker") && t.expireConditionMet()) {
```

Comment:
using string containing to tell if it's endtimechecker is tricky. can we have a better way to handle this?